### ISSUE TITLE
support sasl auth properties options to kafka connector

### ISSUE THREAD

**Issue by davidkl97** [prestosql/presto#6439]
Most of the production deployments of kafka clusters are using sasl authentication mechanism.
it would be nice to add this properties to the connector for production usage.

The properties should contain the mechanism usage, username and password etc for both the Client, Producer and Conumer.

Thanks.

**Comment by kokosing** [comment]
Is this duplicate of https://github.com/prestosql/presto/issues/1184? Or do you mean different SASL mechanisms?

**Comment by davidkl97** [comment]
I  meant SASL Plaintext mechanism. Those mentioned issues relate to Kerbereos or SSL mechanisms...

I guess all of them are relevant for production usage and should be taken for consideration in the end.

**Comment by ashishmgofficial** [comment]
Hi @klDen , Is this still in active development ? We also had this requirement for SASL_SSL configurations . I have also raised a ticket for the same with expected couple of configs : 
```
kafka.sasl.jaas.config
kafka.security.protocol=SASL_SSL
kafka.sasl.mechanism=PLAIN
kafka.sasl.username
kafka.sasl.password
kafka.group.id
basic.auth.credentials.source
basic.auth.user.info

```

**Comment by klDen** [comment]
Hello @ashishmgofficial , 

I was planning on working on this, but according to @lukasz-walkiewicz , a security team member at Starburst will be taking care of this :-)!

**Comment by ashishmgofficial** [comment]
@lukasz-walkiewicz Can you share any info on this ? Wanted to know the expected time this feature would be available to Trino and Starburst

**Comment by kokosing** [comment]
One of the ideas would be to implement `kafka.config.resources` (like `hive.config.resources`) where user could pass their own properties that could be then used when connecting to Kafka services.

**Comment by kokosing** [comment]
Yes. However Trino might override them. For example if you configure SSL in such file and SSL specific properties then they can be overriden.

**Comment by klDen** [comment]
Great. If this is the chosen solution, I'd be happy to help out on this.

**Comment by kokosing** [comment]
Please do.

**Comment by VipinShreyasKumar** [comment]
@kokosing would you be able to point to an implementation snippet of kafka.config.resources usage with Trino. Our kafka cluster is configured with SASL_SSL at the moment.

**Comment by devsinghnaini** [comment]
yes we are also facing issue while connecting to kafka cluster which is SASL_SSL enabled from Trino. 
@kokosing  or @klDen Can you please help us on this configuration for kafka.config.resources or when will this be implemented in Trino.

**Comment by kokosing** [comment]
It was an idea how this could be implemented. It is not yet implemented.

**Comment by klDen** [comment]
I'll start implementing this feature this weekend!

**Comment by gsvlad** [comment]
hi guys. I would like to follow up a discussion.
In our project we want to stream data into Confluent topics from Trino, and Confluent uses SASL authentication method.
What is the current status of implementation of this feature? thanks

**Comment by klDen** [comment]
Hey @gsvlad , I have been busy on my end lately and didn't spend more time on this feature implementation.
What's currently missing are some Unit tests as well as fixing the CI build.
If there are still interest in this feature, I'll try resuming my work on this asap.

**Comment by gsvlad** [comment]
This is much appreciated. I confirm the need as currently SASL authentication is the only supported by Confluent. Thank you!

**Comment by pichlerpa** [comment]
Hi everyone, I would also be interested in that feature, seems to be part of Starburst already? https://docs.starburst.io/latest/connector/starburst-kafka.html#sasl-authentication

**Comment by gsvlad** [comment]
Hey @klDen, are there any updates? We are planning our further activities and some are related to this functionality. I would appreciate if you share your plans about it. Thank you

**Comment by klDen** [comment]
Hey @gsvlad , I'll resume my work here this week!

**Comment by ZZa** [comment]
@klDen, thanks for taking care of this feature!

Just wanted to point out that we also are waiting for it (being able to either propagate our own security configuration for the Kafka connection or being able to specify `SASL_PLAIN` as a security mechanism together with username/password/JAAS config).

And I can confirm that in my experience most productive Kafka clusters used `SASL_PLAIN` or `SASL_SSL` as their security mechanisms. 

Also, Schema Registry connection usually required authentication, so, as @ashishmgofficial proposed, having configuration properties like these ones is also necessary:
```
kafka.confluent-schema-registry-use-ssl=[true|false]
kafka.basic-auth-credentials-source=[URL|USER_INFO]
kafka.basic-auth-user-info=[username:password]
```

Thanks again!

**Comment by matt12eagles** [comment]
Someone posted in 2019 the ability to do this in the slack. (that and dynamically pull over topics).

Would still enjoy this functionality as well. 

Thanks!

**Comment by ZZa** [comment]
> Someone posted in 2019 the ability to do this in the slack.
@matt12eagles, you mean, this should be already possible to propagate custom security configuration to `trino`? Would you be able probably to elaborate on it?

**Comment by ZZa** [comment]
@matt12eagles:
> I had read in slack that someone had got it working w/ a custom fork..

Please share the link to this fork, if you happen to find it here on GitHub.. This would be useful probably also for @klDen to simplify the effort.

**Comment by matt12eagles** [comment]
I don't have a link available, but any update if this is working or if anyone is working on this? May be able to take a shot at it if no one is working on this at the moment

**Comment by klDen** [comment]
Hey! Please try building from my PR branch. Im only missing product test in the PR, but it should be working and allow users to provide more custom Kafka configs. I'll work on the product tests very soon

**Comment by ZZa** [comment]
Greetings in the new year, @klDen! Any updates on including this enhancement in the official build probably?

**Comment by klDen** [comment]
Hello! I'll be resolving the comments this weekend! :)

**Comment by ZZa** [comment]
@klDen, any good/bad news probably? Sorry for bothering you..

**Comment by klDen** [comment]
> @klDen, any good/bad news probably? Sorry for bothering you..

Hey! There's no news ATM. I'm waiting for the maintainers to review my PR.

**Comment by ZZa** [comment]
@klDen, thanks for the info, didn't notice the PR at first: https://github.com/trinodb/trino/pull/8743

Could we link it with this issue, probably?

**Comment by klDen** [comment]
I don't have the permission to link the PR here. People can refer to your comment and see the PR though :+1:

**Comment by kokosing** [comment]
I added link.

**Comment by ZZa** [comment]
@klDen hooray, the review was done! :)

**Comment by klDen** [comment]
> @klDen hooray, the review was done! :)

Hey! I'll work on this end of next week. I'm currently AFK 😄
